### DISCUSSION TITLE
[Feature request] 1password integration

### DISCUSSION THREAD

**Discussion by AdnanTheExcellent** [anonaddy/anonaddy#366] [Feature Requests]
1password announced recently that they teamed with fastmail to automatically generate anonymous passwords when adding a new login to 1password. 

https://1password.com/fastmail/

With 1password you can now create your own custom integrations. It would be awesome if 1password and anonaddy could integrate with each other to save time when creating new accounts.

**Comment by ccorneli** [comment]
This would be awesome. I'm a user of 1P and didn't know that was an offering. I would love to see the integration.

However I see some challenges: 
mainly that You can only enable it by logging into the 1P website and fastmail is the only selection.
So it looks entirely outside of AA's control and dependent on 1P to develop and support it,
But maybe they would respond to @willbrowningme if he reached out to them and find out the details.

I would also guess there was some funding agreement for 1P and FastMail to work together, and if fastmail did any significant development, they might even want some exclusivity for a while.

Finally, I'm not familiar with fastmail, but the integration seems to hook into access of fastmail mailboxes where your email is stored... That might be a 'feature' for most, but but AA thankfully doesn't save a copy of your email... so hopefully that isn't a requirement.  It should be straightforward to  generate a prefix for any domain by some rule.  and then track it as a generated email address.

**Comment by AdnanTheExcellent** [comment]
@ccorneli while it probably wont be integrated within 1password without them getting involved, anonaddy could integrate 1password into their platform. There's [SDKs](https://1password.com/secrets/integrations/) that can connect 1password where anonaddy could not only generate an email but also generate a password and save the login within 1password with their SDK. I know this is fairly large feature thats out of scope but it would save me a lot of time when creating a new login. I'm not sure how many people who use anonaddy also have to populate a password manager, but would convert a several step process of generating a login into a 1 step process.
